### DISCUSSION TITLE
Cypress objects/variables Aliases cannot be accessed using the `this` keyword

### DISCUSSION THREAD

**Discussion by jgonzalezalbisu** [cypress-io/cypress#22294] [Questions and Help] [answered]
After reading this doc https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Sharing-Context I've tried to do this:
```
describe("tests", () => {
    beforeEach("beforeach", () => {
        cy.visit("https://www.google.com/");
        cy.wrap("one").as("miNumero");
    });

    it("test 1", () => {
        expect(this.miNumero).to.equal("one");
    });
});
```
And got this:
![image](https://user-images.githubusercontent.com/23585745/173468365-6418ef7c-f379-4a8b-8fc2-db5b19ab73ab.png)

Same thing for:
```
describe("pruebas", () => {

    beforeEach("beforeach", () => {
        cy.visit("https://www.google.com/");
        cy.fixture("/somefixture").as("users");
    });

    it("prueba2", () => {
        cy.log(users[0]);
    });
});
```

![image](https://user-images.githubusercontent.com/23585745/173468299-41a11b0f-0581-4a93-b708-b38cbbda6825.png)

I've tried also with passing objects / arrays, same thing of course, but with no success.

Any ideas why this is failing?

**Comment by jjhelguero** [comment, marked as answer]
In order to use the `this` keyword, you'll have to use `function() {...}` callbacks. The [docs](https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Sharing-Context) explain more about it.

```js
describe("tests", () => {
    beforeEach("beforeach", () => {
        cy.visit("https://www.google.com/");
        cy.wrap("one").as("miNumero");
    });

    it("test 1", function() {
        expect(this.miNumero).to.equal("one");
    });
});
```

**Reply by jgonzalezalbisu** [reply]
Thank you @jjhelguero!
